python-lib/gluonts_forecasts/training_session.py

```diff
@@ -269,11 +269,10 @@ def _check_timeseries_identifiers_columns_types(self):
     def _compute_optimal_num_batches_per_epoch(self):
         """ Compute the optimal value of num batches which garanties (statistically) full coverage of the dataset """
         sample_length = 2 * self.prediction_length  # Assuming that context_length = prediction_length
-        sample_offset = max(sample_length // 10, 1)  # offset of 1 means that 2 samples can overlap on all but 1 timestep
         num_samples_total = 0
         for timeseries in self.evaluation_train_list_dataset.list_data:
             timeseries_length = len(timeseries[TIMESERIES_KEYS.TARGET])
-            num_samples = (timeseries_length - sample_length) // sample_offset + 1
+            num_samples = math.ceil(timeseries_length / sample_length)
             num_samples_total += num_samples
         optimal_num_batches_per_epoch = math.ceil(num_samples_total / self.batch_size)
```

Comment:
@StanislasGuinel can we add logging?

Reply:
you mean log `optimal_num_batches_per_epoch` before we do `max(optimal_num_batches_per_epoch, 50)`?

Reply:
yes, log optimal_num_batches_per_epoch with a nice message

Reply:
done
145e8c0